src/tpm/kmyth_seal_unseal_impl.c

```diff
@@ -143,93 +140,96 @@ int tpm2_kmyth_seal(uint8_t * input,
   // can then incorporate this result into the objects we create as the
   // authorization policy digest value that must be regenerated to authorize
   // use of these objects.
-  TPM2B_DIGEST objAuthPolicy;
+  //
+  // Note: For now we will not consider a policy-OR authorization criteria.
+  //       We will compute a policy digest based on the "current" authVal
+  //       and/or PCR selection settings. This will be the digest for the
+  //       first policy "branch" in a policy-OR (or the only "branch" if a
+  //       policy-OR criteria is not specified).
 
-  objAuthPolicy.size = 0;
-  if (create_policy_digest(sapi_ctx, ski.pcr_list, &objAuthPolicy))
+  TPM2B_DIGEST objAuthPolicy = {.size = 0, };
+
+  if (create_policy_digest(sapi_ctx,
+                           &(ski.pcr_sel.pcrs[0]),
+                           NULL,
+                           &objAuthPolicy))
   {
-    kmyth_log(LOG_ERR,
-              "error creating policy digest for new Kmyth object ... exiting");
+    kmyth_log(LOG_ERR, "error creating authPolicy digest");
 
     // clear potential 'auth' data, free TPM resources before exiting early
     kmyth_clear(objAuthVal.buffer, objAuthVal.size);
     kmyth_clear(ownerAuth.buffer, ownerAuth.size);
     free_tpm2_resources(&sapi_ctx);
     return 1;
   }
+  kmyth_log(LOG_DEBUG, "created policy digest using current auth/PCR values");
+
+  // There is a command-line argument allowing user to receive a hex dump of
+  // the policy digest. This supports obtaining the digest values to supply
+  // for a policy-OR authorization criteria.
 
-  // optional argument allowing user to receive a hex dump of the policy digest,
-  // to be used to calculate a second policy for policyOR authorization
   if (bool_trial_only == true)
   {
-    // size of string is 2x chars for the size of the TPM2B_DIGEST buffer + 4
-    // for TPM2B struct which encodes size in memory + 1 byte for null termination.
-    // prints to the console and finishes the program without sealing
-    // By using TPM2B_DIGEST size we're reserving a little bit more than needed but
-    // then the output_string can be reserved based on compile-time values.
-    size_t string_size = (2 * sizeof(TPM2B_DIGEST)) + 1;
-    char output_string[string_size];
+    // create stack variable for digest hex-string
```

Comment:
Why does it matter that it's on the stack? I have no objection, just not sure why that's called out in the comment.

Reply:
Yes, it really does not matter. Since it is a temporary buffer that exists only to facilitate printing out the hex-string representing the policy digest, I decided that a stack variable, that didn't require explicit memory allocation and freeing would be a good choice here, but, it is certainly not a required or important that this variable is implemented that way. I have reworded the comment to attempt to emphasize instead that it is a temporary variable (i.e., only exists within the scope of the if statement).